src/txrequest.cpp

```diff
@@ -306,23 +306,28 @@ GenTxid ToGenTxid(const Announcement& ann)
 }  // namespace
 
 /** Actual implementation for TxRequestTracker's data structure. */
-class TxRequestTracker::Impl {
+class TxRequestTracker::Impl
```

Comment:
I know this question is not in the scope of this PR but appreciate the answer. Why there is a separation between `TxRequestTracker` and it's implementation? The comments says: ` Avoid littering this header file with implementation details.`
Is the clarity and readability of code, the whole point of this separation? If yes, is this a common design pattern?

Reply:
That's exactly the point. It's called the "pimpl pattern" (pointer to impl class) -- https://cpppatterns.com/patterns/pimpl.html

Reply:
We also had a review club on this a while ago, if you are interested: https://bitcoincore.reviews/22950

Reply:
Yeah, also here: https://en.cppreference.com/w/cpp/language/pimpl "This technique is used to construct C++ library interfaces with stable ABI and to reduce compile-time dependencies."

What it does for us in practice is to reduce _re_-compilation times - if everything is compiled and some private member is changed then only the `cpp` file will be recompiled instead of all files that include the `h` file.

Improving clarity and readability is not mentioned anywhere in any of the pimpl articles I checked.